src/index/silentpaymentindex.cpp

```diff
@@ -28,8 +29,9 @@ class SilentPaymentIndex::DB : public BaseIndex::DB
     bool WriteSilentPayments(const std::pair<uint256, std::vector<CPubKey>>& tweaks);
 };
 
+// TODO: check if we're on mainnet, otherwise start_height=0
 SilentPaymentIndex::DB::DB(size_t n_cache_size, bool f_memory, bool f_wipe) :
-    BaseIndex::DB(gArgs.GetDataDirNet() / "indexes" / "silentpaymentindex", n_cache_size, f_memory, f_wipe)
+    BaseIndex::DB(gArgs.GetDataDirNet() / "indexes" / "silentpaymentindex", n_cache_size, f_memory, f_wipe, /*start_height=*/TAPROOT_MAINNET_ACTIVATION_HEIGHT)
```

Comment:
This isn't the right place to pass the `start_height`, this is actually setting the obfuscate option of the DB that is constructed here.

Reply:
The `start height` goes into the `SilentPaymentIndex` constructor on line 45 below:

```
SilentPaymentIndex::SilentPaymentIndex(std::unique_ptr<interfaces::Chain> chain, size_t n_cache_size, bool f_memory, bool f_wipe)
    : BaseIndex(std::move(chain), "silentpaymentindex", /*start_height=*/TAPROOT_MAINNET_ACTIVATION_HEIGHT), m_db(std::make_unique<SilentPaymentIndex::DB>(n_cache_size, f_memory, f_wipe))
{}
```

Reply:
Done